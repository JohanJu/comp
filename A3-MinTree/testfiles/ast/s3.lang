int a() {
	int x;
	int y;//sc
	x = 5 * 5;
	y = x;
	x = 5 + 5;
	y = 3 * 4 + 5 * 6;
}
