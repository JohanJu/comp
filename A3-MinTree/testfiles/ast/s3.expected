Program
  List
    Func ID="a"
      List
      List
        Decl ID="x"
          Opt
        Decl ID="y"
          Opt
        Assi ID="x"
          Mul
            Num NUM="5"
            Num NUM="5"
        Assi ID="y"
          IdUse ID="x"
        Assi ID="x"
          Add
            Num NUM="5"
            Num NUM="5"
        Assi ID="y"
          Add
            Mul
              Num NUM="3"
              Num NUM="4"
            Mul
              Num NUM="5"
              Num NUM="6"
